src/System.Data.Common/src/System/Data/Common/DbCommand.cs

```diff
@@ -200,7 +200,7 @@ public virtual Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToke
 
                 try
                 {
-                    return Task.FromResult<int>(ExecuteNonQuery());
+                    return Task.Run<int>(() => ExecuteNonQuery());
```

Comment:
Hi Stephen, Thank you for the response.

I understand the first two points, i have moved the try/finally to inside the Task and i am disposing of the CancellationToken properly.

Still i don't understand your third point, why does accessing the DbCommand method and the cancellation token from the task end up with extra allocation?

In this case you are allocating a Tuple that you later have to unbox from an object.

I would appreciate if you elaborate on that point.

Thank you,
Arik